myapp/app/middlewares/maintenance_mode_filter.rb

```diff
@@ -0,0 +1,13 @@
+class MaintenanceModeFilter # rubocop:disable Style/Documentation
+  def initialize(app)
+    @app = app
+  end
+
+  def call(env)
+    if File.exist?(Rails.root.join('tmp', 'maintenance_mode_on'))
+      [302, { 'Location' => '/maintenance' }, []]
+    else
+      @app.call(env)
```

Comment:
ruby では以下のように後置ifを使ったコーディングがよく見られます。
読みやすくなりますのでぜひご活用ください 👍 

```suggestion
    return [302, { 'Location' => '/maintenance' }, []] if File.exist?(Rails.root.join('tmp', 'maintenance_mode_on'))
    @app.call(env)
```

Reply:
なるほど